### ISSUE TITLE
No way to specify -Wall with MSVC without incurring " WARNING: Consider using the built-in warning_level option"

### ISSUE THREAD

**Issue by CorrodedCoder** [mesonbuild/meson#10796]
**Describe the bug**
It does not seem possible to influence Meson add the "-Wall" flag when compiling with Microsoft Visual C++, and so it must be added manually such as by:
    add_project_arguments('-Wall', language : ['c', 'cpp'])
This is fair enough since this is rarely used with MSVC (although see comments later on this) but doing so generates the output:
    ..\meson.build:11: WARNING: Consider using the built-in warning_level option instead of using "-Wall".
I think that this check makes sense when for toolchains where warning_level can influence this, but since Meson won't add this for MSVC it presents the problem.

Probably irrelevant detail:
Although the issue is minor because -Wall has historically been rarely used with MSVC (in fact it was barely practical to do so), Microsoft have now provided (since VS2015, and getting better with each release since then) the external options which can be used to set the warning level for external headers through various means. See https://docs.microsoft.com/en-us/cpp/build/reference/external-external-headers-diagnostics?view=msvc-170
For example (the first option is only needed prior to Visual Studio 2019 version 16.10):
    '-experimental:external', '-external:W0', '-external:anglebrackets'
These options now make using -Wall a practical possibility and indeed my projects now pretty much always enable them to highly beneficial effect.

**To Reproduce**
meson.build:
```
project('Tester', 'cpp',
	meson_version : '>= 0.63.0',
	default_options : [ 'warning_level=3']
	)

cpp_compiler = meson.get_compiler('cpp')
cpp_compiler_version = cpp_compiler.version()

if cpp_compiler.get_id() == 'msvc' and cpp_compiler_version.version_compare('>=19.00')
	message('Detected Modern Visual Studio: configuring maximum warnings')
	add_project_arguments('-Wall', language : ['c', 'cpp'])
endif

executable('test_a', sources: 'test.cpp')
```

**Expected behavior**
Until such time as Meson provides the mechanism to enable -Wall under MSVC (it's probably still not mature enough yet) it should not complain with MSVC that the option should be set using the built-in warning_level option when it cannot.

**system parameters**
* native build (for the same computer)
* Windows 10
* Python version 3.10.1
* meson --version: 0.63.0
* ninja --version: 1.10.2.git.kitware.jobserver-1

**Comment by eli-schwartz** [comment]
https://github.com/mesonbuild/meson/blob/167356aeabe576b0c777b329748d314d4931a777/mesonbuild/interpreter/interpreter.py#L2798-L2803

Interestingly, we explicitly check for /Wall even though that doesn't correspond to a builtin `-Dwarning_level=` value.

**Comment by CorrodedCoder** [comment]
Yes, quite so. With respect to -Wall vs /Wall, MSVC (as well as most command line tools) for a long time now also accepts hypen's instead of forward slashes, so at least for my sanity I tend to use hypenated options which means I don't need to do anything silly for cross platform defines, for example.
